server/py/services/api/tests/unit/api/test_projects.py

```diff
@@ -289,6 +309,34 @@ def _list_configmaps(*args, **kwargs):
         == {}
     )
 
+    # check that alerts cache is cleaned up
+    for alert_id in alert_ids_to_remove:
+        assert (
+            services.alerts.crud.Alerts()._get_alert_by_id_cached()(db, alert_id)
+            is None
+        )
+        with pytest.raises(mlrun.errors.MLRunNotFoundError):
+            assert services.alerts.crud.Alerts()._get_alert_state_cached()(db, alert_id)
+
+    # check that alerts cache is not cleaned up
+    for alert_id in alert_ids_to_keep:
+        assert (
+            services.alerts.crud.Alerts()._get_alert_by_id_cached()(db, alert_id)
+            is not None
+        )
+        assert (
+            services.alerts.crud.Alerts()._get_alert_state_cached()(db, alert_id)
+            is not None
+        )
+
+    # check that event cache is removed for project_to_remove, and it isn't for project_to_keep
+    found_project_to_keep = False
+    for key in services.alerts.crud.events.Events()._cache.keys():
+        assert key[0] != project_to_remove
+        found_project_to_keep |= key[0] == project_to_keep
```

Comment:
if you know how many events you have for this project you can keep a counter and be more accurate

Reply:
@alonmr which counters? this is just objects cache..

Reply:
I mean do something like
```
    project_to_keep_cached_events_count = 0
    for key in services.alerts.crud.events.Events()._cache.keys():
        assert key[0] != project_to_remove
        if key[0] == project_to_keep:
           project_to_keep_cached_events_count += 1

   assert project_to_keep_cached_events_count == expected_project_to_keep_cached_events_count
```

Reply:
@alonmr done